api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -917,102 +935,135 @@ public String uploadMappings(MultipartFile mapping) {
 		
 		return msg;
 	}
-	
+
 	@Override
-	public String[] exportSelectedMappings(String[] selectedMappings) {
-		String[] cleanedSelectedMappings = cleanSelectedMappings(selectedMappings);
+	public String[] exportMappings(String[] mappings) throws IOException {
```

Comment:
This method is a bit lengthy. Shall we break this into sub methods to make it more readable?

Reply:
I'll see what I can do but it's a bit hard to break this down

Reply:
@jayasanka-sack i broke it down, please review

Reply:
Looks good to me!